lib/gem-miner/github_client.rb

```diff
@@ -0,0 +1,38 @@
+require 'base64'
+require 'octokit'
+
+module GemMiner
+  # A Github client with GemMiner-specific extensions.
+  class GithubClient
+
+    def initialize(github_access_token = nil)
+      @internal_client = Octokit::Client.new(
+        auto_paginate: true,
+        access_token: github_access_token
+      )
```

Comment:
If you don't, it's fine - any public project can be accessed without an access token.